src/modules/java.rs

```diff
@@ -54,12 +55,18 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 }
 
 fn get_java_version(context: &Context) -> Option<String> {
-    let java_command = match context.get_env("JAVA_HOME") {
-        Some(java_home) => format!("{}/bin/java", java_home),
-        None => String::from("java"),
-    };
-
-    let output = context.exec_cmd(&java_command.as_str(), &["-Xinternalversion"])?;
+    let java_command = context
+        .get_env("JAVA_HOME")
+        .and_then(|java_home| {
+            Path::new(&java_home)
+                .join("bin")
+                .join("java")
+                .to_str()
+                .map(str::to_owned)
+        })
+        .unwrap_or(String::from("java"));
```

Comment:
I just tested something similar locally:
```
    let java_command = context
        .get_env("JAVA_HOME")
        .map(PathBuf::from)
        .and_then(|path| {
            path.join("bin")
                .join("java")
                .into_os_string()
                .into_string()
                .ok()
        })
        .unwrap_or("java".to_string());
```

Honestly, I'm not that proficient in Rust to figure out what's better so just pasting it here for you to decide :shrug:

Reply:
I think using `into` might avoid extra allocations.

Reply:
🤷🏻 I've take a bit of both of your suggestions, thanks for your help.

Reply:
I've added a few more suggestions but I guess they are mostly cosmetic at this point so feel free to ignore them. :upside_down_face: